lib/r10k/git/rugged/base_repository.rb

```diff
@@ -1,10 +1,14 @@
 require 'r10k/git/rugged'
+require 'r10k/git/rugged/credentials'
 require 'r10k/logging'
 
 class R10K::Git::Rugged::BaseRepository
 
   include R10K::Logging
 
+  #  @return [Pathname] The path to this repository.
+  attr_reader :path
```

Comment:
nitpick: It's slightly confusing to have @path defined here but set up in the initialization of subclasses. Putting an initialize here seems like spaghetti though since it won't do anything. Perhaps a comment reminding us to set @path in any new subclasses?